tests/benchmark/configs/parser.py

```diff
@@ -0,0 +1,487 @@
+#!/usr/bin/env python3
+"""
+Parser to analyze benchmark tests and maintain the opcode counts mapping.
+
+This script uses Python's AST to analyze benchmark tests and generate/update
+the SCENARIO_CONFIGS mapping in fixed_opcode_counts.py.
+
+Usage:
+    uv run python parser.py --check    # Check for new/missing entries (CI)
+    uv run python parser.py --dry-run  # Show config without changes
+    uv run python parser.py --update   # Update fixed_opcode_counts.py
+
+Conventions:
+    - If test has no opcode parametrization: "test_<name>.*"
+    - If test is parametrized by 'opcode' or 'op': "test_<name>.*<OPCODE>.*"
+"""
+
+import argparse
+import ast
+from pathlib import Path
+from typing import Any
+
+BENCHMARK_DIR = Path(__file__).parent.parent
+CONFIG_FILE = Path(__file__).parent / "fixed_opcode_counts.py"
+
+
+class OpcodeExtractor(ast.NodeVisitor):
+    """Extract opcode parametrizations from benchmark test functions."""
+
+    def __init__(self, source_code: str, filter_code_gen: bool = True):
+        self.source_code = source_code
+        self.patterns: list[str] = []
+        self.filter_code_gen = filter_code_gen
+
+    def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
+        """Visit function definitions and extract opcode patterns."""
+        if not node.name.startswith("test_"):
+            self.generic_visit(node)
+            return
+
+        # Check if function has benchmark_test parameter
+        if not self._has_benchmark_test_param(node):
+            self.generic_visit(node)
+            return
+
+        # Optional: Filter for code generator usage
+        if self.filter_code_gen and not self._uses_code_generator(node):
+            self.generic_visit(node)
+            return
```

Comment:
I’m not sure whether we need `self.generic_visit(node)` in each of the if–else branches. My understanding is that once a condition matches, it will recursively visit the nested function anyway.

This seems relevant only for cases like:
```python
def test_function1(...):

    def test_function2(...):
        ...
```

If so, maybe we could remove the logic here:


```suggestion
        if not node.name.startswith("test_"):
            return

        # Check if function has benchmark_test parameter
        if not self._has_benchmark_test_param(node):
            return

        # Optional: Filter for code generator usage
        if self.filter_code_gen and not self._uses_code_generator(node):
            return
```